common/base/src/logging.cpp

```diff
@@ -27,66 +28,72 @@ namespace genomeworks
 
 namespace logging
 {
-static std::shared_ptr<spdlog::logger> logger = nullptr;
+static std::unique_ptr<std::ostream> out_stream_ = nullptr;
+static std::unique_ptr<std::ofstream> out_file_  = nullptr;
 
-LoggingStatus Init(const char* filename)
+static LogLevel level_ = LogLevel::ERROR;
+
+void check_logger()
 {
-    // for now, first call wins:
-    if (logger != nullptr)
-        return LoggingStatus::success;
+    if (out_stream_ == nullptr)
+    {
+        std::cerr << "GenomeWorks logger not initialized yet. Initializing default logger now." << std::endl;
+        create_logger(LogLevel::ERROR);
+    }
+}
+
+std::string log_level_str(LogLevel level)
+{
+    std::string prefix;
+    switch (level)
+    {
+    case CRITICAL: prefix = "CRITICAL"; break;
+    case ERROR: prefix = "ERROR"; break;
+    case WARN: prefix = "WARN"; break;
+    case INFO: prefix = "INFO"; break;
+    case DEBUG: prefix = "DEBUG"; break;
```

Comment:
Optional: I wouldn't let this function throw on unknown log level, because it is not a critical error - the program will run just fine - and it is unnecessary overhead for an error that should not happen for a bug-free program.
I would make it
```
default:
   assert(false); // Unknown log level.
   prefix = ""; // or use something else, e.g. "INFO".
   break;
```
You'll see the problem in debug mode and in release mode the error is just ignored.

Reply:
nice - I like this, have updated this section